runtime/server/server.go

```diff
@@ -266,22 +238,12 @@ func (s *Server) HTTPHandler(ctx context.Context, registerAdditionalHandlers fun
 		MaxAge: 60 * 60,
 	}
 
-	// Wrap mux with CORS middleware
-	handler := cors.New(corsOpts).Handler(httpMux)
+	// Wrap mux with CORS middleware and trace middleware
+	handler := middleware.TraceMiddleware(cors.New(corsOpts).Handler(httpMux))
 
 	return handler, nil
 }
 
-// HTTPErrorHandler wraps gateway.DefaultHTTPErrorHandler to map gRPC unknown errors (i.e. errors without an explicit
-// code) to HTTP status code 400 instead of 500.
-func HTTPErrorHandler(ctx context.Context, mux *gateway.ServeMux, marshaler gateway.Marshaler, w http.ResponseWriter, r *http.Request, err error) {
-	s := status.Convert(err)
-	if s.Code() == codes.Unknown {
-		err = &gateway.HTTPStatusError{HTTPStatus: http.StatusBadRequest, Err: err}
```

Comment:
I saw you fixed this in `mapGRPCError` in `admin/server`, but the same handling needs to be copied to `mapGRPCError` in this file (`runtime/server`).

I notice that in this file, it's missing this check, which `admin/server` has. You'll probably need to add that here as well:
```
	if _, ok := status.FromError(err); ok {
		return err
	}
```